src/api/v2/quickPickWizard/GenericQuickPickStep.ts

```diff
@@ -0,0 +1,65 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep, IAzureQuickPickItem, NoResourceFoundError } from '@microsoft/vscode-azext-utils';
+import * as vscode from 'vscode';
+import { ResourceModelBase } from '../v2AzureResourcesApi';
+import { ContextValueFilter, matchesContextValueFilter } from './ContextValueFilter';
+import { QuickPickWizardContext } from './QuickPickWizardContext';
+
+export class GenericQuickPickStep<TModel extends ResourceModelBase> extends AzureWizardPromptStep<QuickPickWizardContext<TModel>> {
+    public supportsDuplicateSteps = true;
+
+    public constructor(
+        protected readonly treeDataProvider: vscode.TreeDataProvider<TModel>,
+        protected readonly contextValueFilter: ContextValueFilter,
+    ) {
+        super();
+    }
+
+    public async prompt(wizardContext: QuickPickWizardContext<TModel>): Promise<void> {
+        const selected = await wizardContext.ui.showQuickPick(await this.getPicks(wizardContext), { /* TODO: options */ });
+        wizardContext.currentNode = selected.data;
+    }
+
+    public shouldPrompt(_wizardContext: QuickPickWizardContext<TModel>): boolean {
+        return true;

```

Comment:
I was thinking we may be able to skip the prompt step if:

1. This isn't a matching context value
2. AND There is only one available option
3. OR the user has specified what should be selected as the currentNode at this point

Reply:
I think it always has to depend on the use case. For example, when choosing a resource to delete, we can _never_ skip a step.

Reply:
Yeah I think I'd want to keep the skipping logic here to a minimum for the sake of simplicity too.

> This isn't a matching context value

This doesn't sound like a case where we could skip. More like give them the option to create whatever we're looking for, or throw a "not found" error, right?

> AND There is only one available option

If there is only one available option, we still may want to show the "+ Create X..." item. I'm also not a huge fan of the "skip if only 1" idea because it could confuse users.

> OR the user has specified what should be selected as the currentNode at this point

Not sure what you mean by this one.